addons/account/tests/account_test_savepoint.py

```diff
@@ -3,23 +3,33 @@
 from odoo.tests.common import Form, SavepointCase
 from odoo.tests import tagged
 
+from contextlib import contextmanager
+from unittest.mock import patch
+
 
 @tagged('post_install', '-at_install')
-class AccountingSavepointCase(SavepointCase):
+class AccountTestInvoicingCommon(SavepointCase):
 
-    # -------------------------------------------------------------------------
-    # DATA GENERATION
-    # -------------------------------------------------------------------------
+    @classmethod
+    def copy_account(cls, account):
+        suffix_nb = 1
+        while True:
+            new_code = '%s (%s)' % (account.code, suffix_nb)
+            if account.search_count([('company_id', '=', account.company_id.id), ('code', '=', new_code)]):
+                suffix_nb += 1
+            else:
+                return account.copy(default={'code': new_code})
 
     @classmethod
-    def setUpClass(cls):
-        super(AccountingSavepointCase, cls).setUpClass()
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass()
 
-        chart_template = cls.env.user.company_id.chart_template_id
-        if not chart_template:
+        if chart_template_ref:
+            chart_template = cls.env.ref(chart_template_ref)
+        else:
             chart_template = cls.env.ref('l10n_generic_coa.configurable_chart_template', raise_if_not_found=False)
```

Comment:
I wonder, with the pr of not installing the generic one all the time if we could "if raise, then take the first one you can find or something" (but it might lead to inconsistent behaviour as well)